### PR TITLE
Cleaned emref module and adapted other to new env variables

### PR DESCRIPTION
**Pull request by amjjbonvin**
Docking example now works including the emref module.
Adapted all previous modules to use the new environment variables (related to issue 
Next step is mdref

PS: The environment variables are printed in the output for each model - must be an old debug statement remaining somewhere in the python code

### CODE REVIEW

**Review by rvhonorato — COMMENTED** [review]
194 files, impossible to check everything, will trust @amjjbonvin also on this one!

File: src/haddock/modules/refinement/emref/__init__.py [hunk 1]
```diff
@@ -108,7 +110,7 @@ def run(self, **params):
                 self.path,
                 self.recipe_str,
                 self.params,
-                ambig_f=self.params.get('ambig', None),
+                ambig=self.params.get('ambig', None),
```
Comment (joaomcteixeira) [inline src/haddock/modules/refinement/emref/__init__.py:113]: This was a good bug!

**Comment by joaomcteixeira** [comment]
> PS: The environment variables are printed in the output for each model - must be an old debug statement remaining somewhere in the python code

taking note of this

**Comment by joaomcteixeira** [comment]
I agree with Rodrigo. Got ahead and merge this PR on your own :rocket: :guitar: :fire:

File: src/haddock/modules/refinement/emref/__init__.py (hunk 1 above)
Comment (joaomcteixeira) [inline src/haddock/modules/refinement/emref/__init__.py:113]: There is a bug here with `ambig` not being `ambig_fname`. I will correct that and inspect also other things. Please wait before merging.

**Review by joaomcteixeira — CHANGES_REQUESTED** [review]
Give me some time to clear some small bugs relating to var names.

**Comment by joaomcteixeira** [comment]
The `ambig_fname` was not being loaded due to an error in the variable name being pulled from the param dictionary `ambig` instead of `ambig_fname`. I am correcting that and working on a way to block future var typing errors at this stage. Var typing errors are already corrected when writing the config.

**Comment by joaomcteixeira** [comment]
@amjjbonvin I corrected the capturing of `ambig_fname` parameter. Before there was an error and the ambig file was **NOT** being loaded into the CNS `.inp` file because `self.params.get('ambig', None)` was being used and masked the bug. Now that `ambig_fname = ""` is part of the `defaults.cfg` this can be rewritten to a normal `self.params['ambig_fname']`.


You may wish to review the calculations again after [this change](https://github.com/haddocking/haddock3/pull/140/commits/0b7678445cf520311a261fea95faa035e1abeb80).

**Comment by amjjbonvin** [comment]
Ok- thanks - is this part of the PR I should merge?

Did you check the other modules, they all seem to use the same construction, i.e. ambig and not ambig_fname…

And it was working.

**Comment by joaomcteixeira** [comment]
Yes, I checked the other modules. All corrected now. The new changes are already part of the of this branch. When the PR it merged the new changes will go along. I was working on #138 from this branch. If you want we can solve that before merging this one. Would be best I think.

**Comment by amjjbonvin** [comment]
Ok - then try to solve it - I won’t update further the modules until this is solved.

**Comment by joaomcteixeira** [comment]
Perfect. Converted to `draft` until #142 is solved.

**Comment by joaomcteixeira** [comment]
Because we closed #142, I think this is safe to merge. Whatever comes after #142 is big enough to hold by itself.

**Review by joaomcteixeira — APPROVED** [review]

**Pull request merged**